WalletWasabi/Helpers/EndPointParser.cs

```diff
@@ -56,41 +56,40 @@ public static bool TryParse(string endPointString, int defaultPort, out EndPoint
 
 				var parts = endPointString.Split(':', StringSplitOptions.RemoveEmptyEntries).Select(x => x.Trim().TrimEnd('/').TrimEnd()).ToArray();
 
-				var isDefaultPortInvalid = !ushort.TryParse(defaultPort.ToString(), out ushort dp) || dp < IPEndPoint.MinPort || dp > IPEndPoint.MaxPort;
-				int port;
 				if (parts.Length == 0)
 				{
 					return false;
 				}
-				else if (parts.Length == 1)
-				{
-					if (isDefaultPortInvalid)
-					{
-						return false;
-					}
-					else
-					{
-						port = defaultPort;
-					}
-				}
-				else if (parts.Length == 2)
+				
+				ushort p;
+				int port;
+
+				switch (parts.Length)
 				{
-					var portString = parts[1];
-					if (ushort.TryParse(portString, out ushort p) && p >= IPEndPoint.MinPort && p <= IPEndPoint.MaxPort)
-					{
+					case 1 when IsValidPort(defaultPort.ToString(), out p):
```

Comment:
If you decide to remove the  `IsValidPort` then change this as follow. Otherwise it is okay.
```suggestion
					case 1 when ushort.TryParse(defaultPort.ToString(), out port):
```

Anyway, you don't need `port` and `p`, use port as out.